### ISSUE TITLE
Rake Task Wish #1 - download external images to local objects folder

### ISSUE THREAD

**Issue by dcnb** [CollectionBuilder/collectionbuilder-csv#6]
Rake task that goes and grabs all the external images in a collection and downloads them to the Objects folder

**Comment by evanwill** [comment]
From CONTENTdm or using some method to configure the API 
or just using what ever links the user produces in a CSV?
Does it rename the files via objectid or something?

**Comment by maehr** [comment]
I agree with @dcnb that serving files locally has several advantages. The most basic version would probably download the files and replace the external links with local ones. The information about the previous location is then lost. Is this what you had in mind?

**Comment by evanwill** [comment]
@maehr you might want to check our recent Rake tasks in [rake-utilities-for-digital](https://github.com/uidaholib/rake-utilities-for-digital). It is pretty specific to our usecase right now, but the task [migrate_contentdm](https://github.com/uidaholib/rake-utilities-for-digital/blob/main/docs/migrate_contentdm.md) we used to download files out of our CONTENTdm repository to prep for our static version. It would be a bit similar to your idea of a Omeka downloader.

**Comment by evanwill** [comment]
p.s. @dcnb original request is probably best using something like [download_by_csv task](https://github.com/uidaholib/rake-utilities-for-digital/blob/main/docs/download_by_csv.md)

**Comment by evanwill** [comment]
`rake download_by_csv` is now in this project
